### PR TITLE
EN6-117 introduced support for existing Keycloak

### PR DESCRIPTION
**Pull request by ampie**
Had to split up the KeycloakServer custom resource into KeycloakServer (just Keycloak and its database), and EntandoClusterInfrastructure (UserManagementService, EntandoK8Service, and eventually also DigitalExchange)

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: src/main/java/org/entando/kubernetes/model/infrastructure/EntandoClusterInfrastructure.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        this();
+        this.spec = spec;
+    }
+
+    public EntandoClusterInfrastructure(EntandoClusterInfrastructureSpec spec, ObjectMeta metadata, EntandoCustomResourceStatus status) {
+        this(metadata, spec);
+        this.entandoStatus = status;
+    }
+
+    public EntandoClusterInfrastructure(ObjectMeta metadata, EntandoClusterInfrastructureSpec spec) {
```
Comment (Kerruba) [inline src/main/java/org/entando/kubernetes/model/infrastructure/EntandoClusterInfrastructure.java]: Why this constructor has the parameters inverted?

File: src/test/java/org/entando/kubernetes/model/AbstractEntandoAppPluginLinkTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+public abstract class AbstractEntandoAppPluginLinkTest implements CustomResourceTestUtil {
+
+    public static final String MY_APP_NAMESPACE = "my-app-namespace";
+    public static final String MY_PLUGIN_NAMEPSACE = "my-plugin-namepsace";
+    protected static final String MY_PLUGIN = "my-plugin";
+    private static final String MY_APP = "my-app";
+
+    @BeforeEach
+    public void deleteentandoAppPluginLinks() throws InterruptedException {
```
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/model/AbstractEntandoAppPluginLinkTest.java]: rename to deleteEntandoAppPluginLinks

File: src/main/java/org/entando/kubernetes/model/infrastructure/EntandoClusterInfrastructure.java (hunk 1 above)
Comment (ampie) [reply src/main/java/org/entando/kubernetes/model/infrastructure/EntandoClusterInfrastructure.java]: Yeah I wasn't very consistent. Generally parameters should be from the more general to more specific, more static to more dynamic. Following these guidelines, 've changed all the model classes now to the following parameter order:
ObjectMetadata, Spec, Status

File: src/test/java/org/entando/kubernetes/model/AbstractEntandoAppPluginLinkTest.java (hunk 2 above)
Comment (ampie) [reply src/test/java/org/entando/kubernetes/model/AbstractEntandoAppPluginLinkTest.java]: Done

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**